test/views/behaviours/ViewBehaviours.scala

```diff
@@ -23,60 +23,82 @@ trait ViewBehaviours extends ViewSpecBase {
 
   val expectTimeoutDialog: Boolean = true
 
-  protected def normalPage(
-    view: HtmlFormat.Appendable,
-    messageKeyPrefix: String,
-    messageKeySuffix: String,
-    messageHeadingArgs: Any*
-  )(
+  val viewViaApply: HtmlFormat.Appendable
+  val viewViaRender: HtmlFormat.Appendable
+  val viewViaF: HtmlFormat.Appendable
+
+  def normalPage(messageKeyPrefix: String, messageKeySuffix: String, messageHeadingArgs: Any*)(
```

Comment:
Personally, I would make the messageKeySuffix optional as majority of views don't have this and it is more clearer, however if you don't want to change it and since it is mandatory, can this please be defaulted to an empty string as I can see it in a lot of view tests currently, thanks.

Reply:
Wanted to do this but due to the `messageHeadingArgs` being of arbitrary length you can't add defaults directly to parameters. Can make add messageHeadingArgs to its own parameter list but would still need empty parameters defined when calling the method.

Saw the suggestion to make it an Option on the other comments, just not sure if that looks cleaner either.

Reply:
I mean similar was done for the pageByMethodWithAssertions method so not sure what the issue is here. I didn't really put much thought into it when reviewing. Option is **clearer** as it states what it is really. Looking at this bit of code without looking at the actual tests, I would think that all views have messageKeySuffix which is wrong as a lot of views don't which was why I suggested option and I didn't really like seeing the empty string either in most of the tests as most views don't have a suffix. This was why I suggested defaulting in one place so at least you supply the suffix when needed.  Something along the lines in the below

```
def normalPage(messageKeyPrefix: String, messageKeySuffix: String = "")(
    messageHeadingArgs: Any*
  )(expectedGuidanceKeys: String*)
```
obviously this would change as we are removing expectedGuidanceKeys

```
  protected def pageByMethod(
    view: HtmlFormat.Appendable,
    messageKeyPrefix: String,
    messageKeySuffix: String = "",
    byMethod: String = "",
    isError: Boolean = false
  )(messageHeadingArgs: Any*): Unit
```
  
```
    def pageByMethodWithAssertions(
    view: HtmlFormat.Appendable,
    messageKeyPrefix: String,
    messageKeySuffix: String = "",
    byMethod: String = "",
    isError: Boolean = false
  )(messageHeadingArgs: Any*)(assertions: => Unit)
```
obviously this pageByMethodWithAssertions would change as we decoupling

Let me know if you want to go through the option route and we can jump on a call which will be faster.